### PR TITLE
Add option to control accessing external servers

### PR DESCRIPTION
**Pull request by DennisHeimbigner**
Add options
"--enable-external-server-tests" (automake)
"-DENABLE_EXTERNAL_SERVER_TESTS" (cmake)
to enable/disable tests that access
external URLS, where an external URL refers to some server not
directly under the control of netcdf-c.
This option defaults to off.

The reason for this PR is that external servers can be up or down
or change content without any warning to us, thus causing some
of our tests to fail.

Additionally. and recently, the thredds-test server was shut
down and some of our tests used it. So these tests are disabled
until a suitable replacement server is available.

## Misc. Fixes
* Bump the depth of trace support in nclog.c to avoid overflow

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -7,12 +7,14 @@ This file contains a high-level description of this package's evolution. Release
...
+* [Bug Fix] Split the remote tests into two parts: one for the remotetest server and one for all other external servers. Also add a configure option to enable the latter set. See [Github #2491](https://github.com/Unidata/netcdf-c/pull/2491).
+* [Bug Fix] Fix support for reading arrays of HDF5 fixed size strings. See [Github #2462](https://github.com/Unidata/netcdf-c/pull/2466).
 * [Bug Fix] Provide a default enum const when fill value does not match any enum constant for the value zero. See [Github #2462](https://github.com/Unidata/netcdf-c/pull/2462).
 * [Bug Fix] Fix the json submodule symbol conflicts between libnetcdf and the plugin specific netcdf_json.h. See [Github #2448](https://github.com/Unidata/netcdf-c/pull/2448).
 * [Bug Fix] Fix quantize with CLASSIC_MODEL files. See [Github #2405](https://github.com/Unidata/netcdf-c/pull/2445).
 * [Enhancement] Add `--disable-quantize` option to `configure`.
 * [Bug Fix] Fix CMakeLists.txt to handle all acceptable boolean values for -DPLUGIN_INSTALL_DIR. See [Github #2430](https://github.com/Unidata/netcdf-c/pull/2430).
+>>>>>>> master
```
Comment (Micket) [inline RELEASE_NOTES.md:17]: you have managed to merge some git-conflict stuff in several places int he RELEASE_NOTES.md file.

**Comment by WardF** [comment]
@DennisHeimbigner I've tried to resolve the conflicts listed above twice but GitHub has rejected my attempts twice with a strange error.  Are you able to resolve them on your side?

**Comment by DennisHeimbigner** [comment]
Ok, I added a commit that I think fixes the conflicts.

**Review by WardF — APPROVED** [review]

**Comment by WardF** [comment]
Thanks @DennisHeimbigner !

**Pull request merged**